apis/v1alpha2/grpcroute_types.go

```diff
@@ -0,0 +1,445 @@
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of GRPCRoute.
+	Spec GRPCRouteSpec `json:"spec,omitempty"`
+
+	// Status defines the current state of GRPCRoute.
+	Status GRPCRouteStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// GRPCRouteList contains a list of GRPCRoute.
+type GRPCRouteList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []GRPCRoute `json:"items"`
+}
+
+// GRPCRouteStatus defines the observed state of GRPCRoute.
+type GRPCRouteStatus struct {
+	RouteStatus `json:",inline"`
+}
+
+// GRPCRouteSpec defines the desired state of GRPCRoute
+type GRPCRouteSpec struct {
+	CommonRouteSpec `json:",inline"`
+
+	// Hostnames defines a set of hostname that should match against the GRPC
+	// Host header to select a GRPCRoute to process the request. This matches
+	// the RFC 1123 definition of a hostname with 2 notable exceptions:
+	//
+	// 1. IPs are not allowed.
+	// 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
+	//    label must appear by itself as the first label.
+	//
+	// If a hostname is specified by both the Listener and GRPCRoute, there
+	// must be at least one intersecting hostname for the GRPCRoute to be
+	// attached to the Listener. For example:
+	//
+	// * A Listener with `test.example.com` as the hostname matches GRPCRoutes
+	//   that have either not specified any hostnames, or have specified at
+	//   least one of `test.example.com` or `*.example.com`.
+	// * A Listener with `*.example.com` as the hostname matches GRPCRoutes
+	//   that have either not specified any hostnames or have specified at least
+	//   one hostname that matches the Listener hostname. For example,
+	//   `test.example.com` and `*.example.com` would both match. On the other
+	//   hand, `example.com` and `test.example.net` would not match.
+	//
+	// If both the Listener and GRPCRoute have specified hostnames, any
+	// GRPCRoute hostnames that do not match the Listener hostname MUST be
+	// ignored. For example, if a Listener specified `*.example.com`, and the
+	// GRPCRoute specified `test.example.com` and `test.example.net`,
+	// `test.example.net` must not be considered for a match.
+	//
+	// If both the Listener and GRPCRoute have specified hostnames, and none
+	// match with the criteria above, then the GRPCRoute is not accepted. The
+	// implementation must raise an 'Accepted' Condition with a status of
+	// `False` in the corresponding RouteParentStatus.
+	//
+	// If a Route (A) of type HTTPRoute or GRPCRoute is attached to a
+	// Listener and that listener already has another Route (B) of the other
+	// type attached and the intersection of the hostnames of A and B is
+	// non-empty, then the implementation must reject Route A. That is, the
+	// implementation must raise an 'Accepted' condition with a status of
+	// 'False' in the corresponding RouteParentStatus.
+	//
+	// Support: Core
```

Comment:
Just thinking about what it means for GRPC specific stuff to be marked as Core if GRPCRoute itself is Extended

Referring to: https://gateway-api.sigs.k8s.io/concepts/guidelines/#overlapping-support-levels

Since these fields are never embedded in a struct with Core support level, then these fields can never be Core?

So ... it's just a nod to "when GRPCRoute becomes Core, some fields will be Core and some will be Extended" ?

Reply:
No, the idea here is that this feature is Core to the `GRPCRoute` feature. Independent of that, `GRPCRoute` is extended relative to the Gateway API. There was previously a discussion on that topic [here](https://github.com/kubernetes-sigs/gateway-api/pull/1004#discussion_r840945168).